array.go

```diff
@@ -568,3 +570,339 @@ func (a *Array) URI() (string, error) {
+				ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue))
+			}
+		}
+	case reflect.Uint8:
+		datatype = TILEDB_UINT8
+		if isSliceValue {
+			tmpValue := value.([]uint8)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue[0]))
+		} else {
+			tmpValue := value.(uint8)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue))
+		}
+	case reflect.Uint16:
+		datatype = TILEDB_UINT16
+		if isSliceValue {
+			tmpValue := value.([]uint16)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue[0]))
+		} else {
+			tmpValue := value.(uint16)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue))
+		}
+	case reflect.Uint32:
+		datatype = TILEDB_UINT32
+		if isSliceValue {
+			tmpValue := value.([]uint32)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue[0]))
+		} else {
+			tmpValue := value.(uint32)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue))
+		}
+	case reflect.Uint64:
+		datatype = TILEDB_UINT64
+		if isSliceValue {
+			tmpValue := value.([]uint64)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue[0]))
+		} else {
+			tmpValue := value.(uint64)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue))
+		}
+	case reflect.Float32:
+		datatype = TILEDB_FLOAT32
+		if isSliceValue {
+			tmpValue := value.([]float32)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue[0]))
+		} else {
+			tmpValue := value.(float32)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue))
+		}
+	case reflect.Float64:
+		datatype = TILEDB_FLOAT64
+		if isSliceValue {
+			tmpValue := value.([]float64)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue[0]))
+		} else {
+			tmpValue := value.(float64)
+			ret = C.tiledb_array_put_metadata(a.context.tiledbContext, a.tiledbArray, ckey, C.tiledb_datatype_t(datatype), valueNum, unsafe.Pointer(&tmpValue))
+		}
+	case reflect.String:
+		datatype = TILEDB_STRING_UTF8
```

Comment:
Are golang strings UTF8 or ascii?

Reply:
I had the same question, the answer was `In Go, a string is in effect a read-only slice of bytes`

This is a valid string `const sample = "\xbd\xb2\x3d\xbc\x20\xe2\x8c\x98"`
Contains non-ASCII...

In essence based on https://blog.golang.org/strings i believe that UTF8 is the safest choice

Axiomatic details:
    Go source code is always UTF-8.
    A string holds arbitrary bytes.
    A string literal, absent byte-level escapes, always holds valid UTF-8 sequences.
    Those sequences represent Unicode code points, called runes.
    No guarantee is made in Go that characters in strings are normalized.

Reply:
Thanks for the analysis!